src/rt/test/func/bag/bag.cc

```diff
@@ -0,0 +1,87 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+#include "ds/bag.h"
+
+#include "test/harness.h"
+#include "test/log.h"
+#include "verona.h"
+
+using namespace snmalloc;
+using namespace verona::rt;
+
+void test()
+{
+  auto& alloc = ThreadAlloc::get();
+
+  Bag<uintptr_t, uintptr_t, Alloc> bag;
+
+  const int NUM_OBJECTS = 128;
+
+  // Allocate a bunch of objects so that the region vector uses many
+  // allocation blocks to track each object in the region.
+  for (uintptr_t i = 0; i < NUM_OBJECTS; i++)
+  {
+    bag.insert({nullptr, i}, alloc);
+  }
+
+  // Iterate through the bag and make sure that all the
+  // entries make sense.
+  Bag<uintptr_t, uintptr_t, Alloc>::iterator iter(&bag);
+
+  Bag<uintptr_t, uintptr_t, Alloc>::Elem* rm1 = nullptr;
+  Bag<uintptr_t, uintptr_t, Alloc>::Elem* rm2 = nullptr;
+  Bag<uintptr_t, uintptr_t, Alloc>::Elem* rm3 = nullptr;
+
+  uintptr_t count = 0;
+  for (auto entry : iter)
+  {
+    assert(entry->metadata == (NUM_OBJECTS - count - 1));
+
+    if (count == 5)
+      rm1 = entry;
+
+    if (count == 40)
+      rm2 = entry;
+
+    if (count == 100)
+      rm3 = entry;
+    count++;
+  }
+
+  assert(count == NUM_OBJECTS);
+
+  bag.remove(rm3);
+  bag.remove(rm2);
+  bag.remove(rm1);
+
+  // Check that the iterator skips over holes left in the region vector
+  count = 0;
+  for (auto item : iter)
+  {
+    UNUSED(item);
+    count++;
+  }
+  assert(count == NUM_OBJECTS - 3);
```

Comment:
Don't use assert in tests.  This wants to be checked on release builds too.   We have a macro that does `check`s:
```suggestion
  check(count == NUM_OBJECTS - 3);
```